phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -647,7 +657,9 @@ public Pair<Integer, Tuple> call() throws SQLException {
                                         lastState.getUpdateCount());
                                 Result result = null;
                                 if (connection.getAutoCommit()) {
-                                    if (isSingleRowUpdatePlan(isUpsert, isDelete, plan)) {
+                                    boolean singleRowUpdate = isSingleRowUpdatePlan(isUpsert,
```

Comment:
You can enhance this method and check if the plan is ServerSelectDeleteMutationPlan and modify its scan. All the other checks like autocommit and row count are also happening already.